apps/app/src/test/scala/org/lfdecentralizedtrust/splice/integration/tests/UnclaimedActivityRecordIntegrationTest.scala

```diff
@@ -0,0 +1,311 @@
+      .simpleTopology1Sv(this.getClass.getSimpleName)
+
+  override protected lazy val sanityChecksIgnoredRootCreates: Seq[Identifier] = Seq(
+    UnclaimedReward.TEMPLATE_ID_WITH_PACKAGE_ID
+  )
+
+  "An UnallocatedUnclaimedActivityRecord gets expired" in { implicit env =>
+    val sv1Party = sv1Backend.getDsoInfo().svParty
+    val aliceUserName = aliceWalletClient.config.ledgerApiUser
+    val aliceParty = aliceValidatorBackend.onboardUser(aliceUserName)
+    val amountToMint = 15.0
+
+    def allocateTriggers: Seq[Trigger] =
+     activeSvs.map(_.dsoDelegateBasedAutomation.trigger[AllocateUnallocatedUnclaimedActivityRecordTrigger])
+    def expiredUnallocatedTriggers: Seq[Trigger] =
+     activeSvs.map(_.dsoDelegateBasedAutomation.trigger[ExpiredUnallocatedUnclaimedActivityRecordTrigger])
+    setTriggersWithin(
+     triggersToPauseAtStart = allocateTriggers ++ expiredUnallocatedTriggers,
+     triggersToResumeAtStart = Seq.empty,
+    ) {
+     actAndCheck(
+       "Creating vote request", {
+         val action = new ARC_DsoRules(
+           new SRARC_CreateUnallocatedUnclaimedActivityRecord(
+             new DsoRules_CreateUnallocatedUnclaimedActivityRecord(
+               aliceParty.toProtoPrimitive,
+               BigDecimal(amountToMint).bigDecimal,
+               "alice is doing great - vote",
+               Instant.now().plus(5, ChronoUnit.SECONDS),
+             )
+           )
+         )
+         eventuallySucceeds() {
+           sv1Backend.createVoteRequest(
+             sv1Party.toProtoPrimitive,
+             action,
+             "url",
+             "alice is doing great",
+             sv1Backend.getDsoInfo().dsoRules.payload.config.voteRequestTimeout,
+             None,
+           )
+         }
+       },
+     )("vote request and UnallocatedUnclaimedActivityRecord have been created",
+        _ => {
+         sv1Backend.listVoteRequests().loneElement
+          eventually(){
+           sv1Backend.participantClient.ledger_api_extensions.acs
+             .filterJava(UnallocatedUnclaimedActivityRecord.COMPANION)(dsoParty) should not be empty
+          }
+        }
+      )
+    }
+
+    setTriggersWithin(
+     triggersToPauseAtStart = allocateTriggers,
+     triggersToResumeAtStart = Seq.empty,
+    ) {
+      clue("UnallocatedUnclaimedActivityRecord gets archived"){
+        eventually(){
+         sv1Backend.participantClient.ledger_api_extensions.acs
+           .filterJava(UnallocatedUnclaimedActivityRecord.COMPANION)(dsoParty) shouldBe empty
+        }
+      }
+    }
+  }
+
+  "An UnclaimedActivityRecord gets expired" in { implicit env =>
+    val sv1UserId = sv1WalletClient.config.ledgerApiUser
+    val sv1Party = sv1Backend.getDsoInfo().svParty
+    val aliceUserName = aliceWalletClient.config.ledgerApiUser
+    val aliceParty = aliceValidatorBackend.onboardUser(aliceUserName)
+    val amountToMint = 15.0
+    val unclaimedRewardsToMint = Seq(15.0)
+
+    def allocateTriggers: Seq[Trigger] =
```

Comment:
you seem to duplicate these definitions across most tests, can we factor them out